airflow-core/src/airflow/migrations/versions/0041_3_0_0_rename_dataset_as_asset.py

```diff
@@ -101,14 +103,22 @@ def _rename_pk_constraint(
 
 
 def _drop_fkey_if_exists(table, constraint_name):
-    dialect = op.get_bind().dialect.name
-    if dialect == "sqlite":
-        try:
-            with op.batch_alter_table(table, schema=None) as batch_op:
-                batch_op.drop_constraint(op.f(constraint_name), type_="foreignkey")
-        except ValueError:
-            pass
-    elif dialect == "mysql":
+    conn = op.get_bind()
+    dialect_name = conn.dialect.name
+
+    if dialect_name == "sqlite":
+        exitstack = contextlib.ExitStack()
+        # SQLite requires foreign key constraints to be disabled during batch operations
+        conn.execute(text("PRAGMA foreign_keys=OFF"))
+        exitstack.callback(conn.execute, text("PRAGMA foreign_keys=ON"))
+
+        with exitstack:
+            try:
+                with op.batch_alter_table(table, schema=None) as batch_op:
+                    batch_op.drop_constraint(op.f(constraint_name), type_="foreignkey")
+            except ValueError:
+                pass
```

Comment:
```suggestion
        conn.execute(text("PRAGMA foreign_keys=OFF"))
        try:
            with op.batch_alter_table(table, schema=None) as batch_op:
                batch_op.drop_constraint(op.f(constraint_name), type_="foreignkey")
        except ValueError:
            pass
        conn.execute(text("PRAGMA foreign_keys=ON"))
```

is there any reason we don't try something like this ?

Reply:
Updated: